compiler/plugins/target/AMD-AIE/aievec/VectorToVectorConversions.cpp

```diff
@@ -925,6 +927,156 @@ struct CanonicalizeVectorForAIEVecPass
   }
 };
 
+/// Returns one of:
+/// 1) failure, if there is definitely an error that should be propagated.
+/// 2) a new transfer_read operation that is sufficiently aligned, if the old
+///    transfer_read is determined to be insufficiently aligned and it is
+///    possible to create a new transfer_read.
+/// 3) the original transfer_read operation, otherwise.
+///
+/// This is currently hard-wired for AIE2 (phoenix) HW constraints.
+FailureOr<Value> getAlignedTransferRead(vector::TransferReadOp readOp,
+                                        IRRewriter &rewriter) {
+  //`alignBits`: transfer_reads must be aligned to this number of bits,
+  //             otherwise the reads will be scalarized.
+  //
+  //`maxVectorSizeBits`: the maximum size of a vector that can be read in a
+  //                     single transfer_read operation.
+  int alignBits = 256;
+  int maxVectorSizeBits = 1024;
```

Comment:
We should retrieve this from the `AMDAIEDeviceModel`. Ideally, you check whether it can be retrieved from aie-rt and otherwise add it here: https://github.com/nod-ai/iree-amd-aie/blob/c6156e3b203401c25d348ae995fd2f1ffc138787/runtime/src/iree-amd-aie/aie_runtime/iree_aie_runtime.h#L224

Reply:
There's nothing there related to the core, are you sure that's a good place to add it? We already have vector size related stuff elsewhere like matmul instruction size:  https://github.com/nod-ai/iree-amd-aie/blob/c6156e3b203401c25d348ae995fd2f1ffc138787/compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/AMDAIEUtils.cpp#L50

Reply:
Yeah, at least all hardware related information is in one place then. Doing this will make it really easy to support multiple architectures. I know we currently have hardcoded information all over the place, but that will make it really cumbersome/buggy to support multiple/new architectures. All that information, like the map you're referencing, should be moved to the device model.